resources/prosody-plugins/util.lib.lua

```diff
@@ -73,8 +73,67 @@ function wrap_async_run(event,handler)
     return result;
 end
 
+--- Updates presence stanza, by adding identity node
+-- @param stanza the presence stanza
+-- @param user the user to which presence we are updating identity
+-- @param group the group of the user to which presence we are updating identity
+-- @param creator_user the user who created the user which presence we
+-- are updating (this is the poltergeist case, where a user creates
+-- a poltergeist), optional.
+-- @param creator_group the group of the user who created the user which
+-- presence we are updating (this is the poltergeist case, where a user creates
+-- a poltergeist), optional.
+function update_presence_identity(
+    stanza, user, group, creator_user, creator_group)
+
+    -- First remove any 'identity' element if it already
+    -- exists
```

Comment:
reading this now i think it's probably worth tweaking this comment to add "so it cannot be spoofed by a client"

Reply:
I've added the comment.